micro-rdk/src/esp32/certificate.rs

```diff
@@ -28,3 +49,277 @@ impl Certificate for WebRtcCertificate {
+        unsafe {
+            MbedTLSError::to_unit_result(mbedtls_ctr_drbg_seed(
+                &mut self.drbg_context as *mut mbedtls_ctr_drbg_context,
+                Some(mbedtls_entropy_func),
+                &mut self.entropy as *mut _ as *mut c_void,
+                std::ptr::null(),
+                0,
+            ))
+        }?;
+
+        unsafe {
+            MbedTLSError::to_unit_result(mbedtls_pk_setup(
+                &mut self.kp_context as *mut mbedtls_pk_context,
+                mbedtls_pk_info_from_type(mbedtls_pk_type_t_MBEDTLS_PK_ECKEY),
+            ))
+        }?;
+
+        let ecp_keypair = self.kp_context.pk_ctx;
+
+        unsafe {
+            MbedTLSError::to_unit_result(mbedtls_ecp_gen_key(
+                mbedtls_ecp_group_id_MBEDTLS_ECP_DP_SECP256R1,
+                ecp_keypair as *mut _ as *mut mbedtls_ecp_keypair,
+                Some(mbedtls_ctr_drbg_random),
+                &mut self.drbg_context as *mut _ as *mut c_void,
+            ))
+        }?;
+
+        unsafe {
+            mbedtls_x509write_crt_set_subject_key(
+                &mut self.crt_context as *mut mbedtls_x509write_cert,
+                &mut self.kp_context as *mut mbedtls_pk_context,
+            );
+            mbedtls_x509write_crt_set_issuer_key(
+                &mut self.crt_context as *mut mbedtls_x509write_cert,
+                &mut self.kp_context as *mut mbedtls_pk_context,
+            );
+            mbedtls_x509write_crt_set_version(
+                &mut self.crt_context as *mut mbedtls_x509write_cert,
+                2,
+            );
+            mbedtls_x509write_crt_set_md_alg(
+                &mut self.crt_context as *mut mbedtls_x509write_cert,
+                mbedtls_md_type_t_MBEDTLS_MD_SHA256,
+            );
+        };
+
+        let subject_name = {
+            CString::new(format!(
+                "CN={},O={},C={}",
+                self.oid_cn, self.oid_o, self.oid_c
+            ))
+            .unwrap()
+        };
+        unsafe {
+            MbedTLSError::to_unit_result(mbedtls_x509write_crt_set_issuer_name(
+                &mut self.crt_context as *mut mbedtls_x509write_cert,
+                subject_name.as_ptr(),
+            ))
+        }?;
+
+        unsafe {
+            MbedTLSError::to_unit_result(mbedtls_x509write_crt_set_subject_name(
+                &mut self.crt_context as *mut mbedtls_x509write_cert,
+                subject_name.as_ptr(),
+            ))
+        }?;
+
+        let not_before = CString::new(self.not_before.clone()).unwrap();
+        let not_after = CString::new(self.not_after.clone()).unwrap();
+
+        unsafe {
+            MbedTLSError::to_unit_result(mbedtls_x509write_crt_set_validity(
+                &mut self.crt_context as *mut mbedtls_x509write_cert,
+                not_before.as_ptr(),
+                not_after.as_ptr(),
+            ))
+        }?;
+
+        let mut work_buffer = vec![0_u8; 2048];
```

Comment:
- What guarantees that 2048 is sufficient?
- If there is a known upper bound, can we use a constant, or the `work_buffer`'s capacity in the call to `write_crt_der` below?

Reply:
unless we start adding a lot of extension and extra information we should be safe with 2048